spec/features/search_spec.rb

```diff
@@ -30,4 +30,177 @@
+      created_at: 5.years.ago,
+      name: "#{searching_for} Miaus",
+      email_subscriber: true)
+
+    visit admin_customers_path
+    fill_in :search, with: query
+    page.execute_script("$('.search').submit()")
+
+    page.within("tr.table__row", match: :first) do
+      expect(page).to have_content(subscribed_and_old_match.name)
+    end
+    expect(page).not_to have_content(subscribed_unmathed.name)
+    expect(page).not_to have_content(unsubscribed_matched.name)
+    expect(page).not_to have_content(subscribed_match_but_new.name)
+  end
+
+  scenario "admin clicks a scope button defined in an array", :js do
+    subscribed = create(
+      :customer,
+      name: "Lua Miaus",
+      email_subscriber: true)
+    unsubscribed = create(
+      :customer,
+      name: "John Doe",
+      email_subscriber: false)
+
+    visit admin_customers_path
+
+    # Included into the COLLECTION_SCOPES array of the CustomersDashboard
+    click_on "subscribed"
+
+    page.within("tr.table__row", match: :first) do
+      expect(page).to have_content(subscribed.name)
+    end
+    expect(page).not_to have_content(unsubscribed.name)
+  end
+
+  scenario "admin clicks a scope button defined in a hash", :js do
+    address_zip = "00000-9999"
+    searched = create(:order, address_zip: address_zip)
+    # other = create(:order) # not used, explained below
+    create(:order)
+
+    visit admin_orders_path
+
+    # Included into the COLLECTION_SCOPES hash of the OrdersDashboard
+    click_on "zip_prefix(00000)"
+
+    page.within("tr.table__row", match: :first) do
+      expect(page).to have_content(searched.id)
+      expect(page).to have_content(searched.customer.name)
+    end
+    # *id* is not a valid field to check exclusion from results :(
+    # expect(page).not_to have_content(other.id)
+  end
+
+  scenario "admin searches using a model scope w/ an argument", :js do
+    query = "scope:name_starts_with(L)"
+    match = create(
+      :customer,
+      name: "Lua Miaus")
+    unmatch = create(
+      :customer,
+      name: "John Doe")
+
+    visit admin_customers_path
+    fill_in :search, with: query
+    page.execute_script("$('.search').submit()")
+
+    page.within("tr.table__row", match: :first) do
+      expect(page).to have_content(match.name)
+    end
+    expect(page).not_to have_content(unmatch.name)
+  end
+
+  scenario "admin searches using a 'wildcarded' scope", :js do
+    query = "name_starts_with:ZZ"
+    match = create(
+      :customer,
+      name: "ZZTop")
+    unmatch = create(
+      :customer,
+      name: "John Doe")
+
+    visit admin_customers_path
+    fill_in :search, with: query
+    page.execute_script("$('.search').submit()")
+    page.within("tr.table__row", match: :first) do
+      expect(page).to have_content(match.name)
+    end
+    expect(page).not_to have_content(unmatch.name)
+
+    # ...and the wildcarded scope doesn't have its button to be clicked.
+    expect(page).not_to have_content('name_starts_with:*')
```

Comment:
Prefer double-quoted strings unless you need single quotes to avoid extra backslashes for escaping.